pkgs/applications/window-managers/sway/wrapper.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, sway-unwrapped
+, swaybg
+, dbus
+, wrapGAppsHook
+, symlinkJoin
+, writeShellScriptBin
+, withBaseWrapper ? true
+, withGTKWrapper ? false
+, extraSessionCommands ? ""
+}:
+
+with lib;
+
+let
+  swayPackage = sway-unwrapped;
+
+  baseWrapper = writeShellScriptBin "sway" ''
+    set -o errexit
+
+    if [ ! "$_SWAY_WRAPPER_ALREADY_EXECUTED" ]; then
+      export _SWAY_WRAPPER_ALREADY_EXECUTED=1
+      ${extraSessionCommands}
+    fi
+
+    if [ "$DBUS_SESSION_BUS_ADDRESS" ]; then
+      export DBUS_SESSION_BUS_ADDRESS
+      exec ${swayPackage}/bin/sway "$@"
+    else
+      exec ${dbus}/bin/dbus-run-session ${swayPackage}/bin/sway "$@"
+    fi
+  '';
+
+  swayJoined = symlinkJoin {
+    name = "sway-base-wrapped";
+    paths = (optional withBaseWrapper [baseWrapper]) ++ [swayPackage];
+  };
+in
+symlinkJoin {
+  name = "sway";
+
+  paths = [swayJoined];
```

Comment:
This package uses 2x `symlinkJoin` and we don't need to join anything here. It would be great if we would only need to use it once to add the `baseWrapper`.